src/app/components/pages/IsaacCompetition/EntryForm/CompetitionEntryForm.tsx

```diff
@@ -0,0 +1,164 @@
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    const form = event.target as HTMLFormElement;
+    const elements = form.elements as any;
+    const groupId = elements.formGroup.value;
+    const selectedGroup = activeGroups.find((group) => group.groupName === groupId);
+    const submissionLink = elements.submissionLink.value;
+    const groupName = selectedGroup?.groupName;
+
+    if (selectedGroup?.id) {
+      const reservableIds =
+        selectedGroup.members?.map((member) => member.id).filter((id): id is number => id !== undefined) || [];
+      reserveUsersOnCompetition(COMPETITON_ID, reservableIds, submissionLink, groupName);
+    }
+  };
+
+  return (
+    <div className="py-5">
+      <div className="entry-form-background-img entry-form-section">
+        <Container className="pb-2">
+          <Form onSubmit={handleSubmit}>
+            <h1 className="py-4 entry-form-title">Enter the competition</h1>
+            <Row className="d-flex flex-column flex-md-row">
+              <Col lg={6}>
+                {renderFormGroup(
+                  "First Name",
+                  "text",
+                  "firstName",
+                  targetUser?.loggedIn ? targetUser.givenName || "" : "",
+                  [],
+                  true,
+                )}
+                {renderFormGroup(
+                  "Last Name",
+                  "text",
+                  "lastName",
+                  targetUser?.loggedIn ? targetUser?.familyName || "" : "",
+                  [],
+                  true,
+                )}
+                {targetUser && (
+                  <FormGroup>
+                    <Label className="entry-form-sub-title">School</Label>
+                    <SchoolInput
+                      disableInput={true}
+                      userToUpdate={{ ...targetUser, password: null }}
+                      submissionAttempted={false}
+                      required={true}
+                      showLabel={false}
+                    />
+                  </FormGroup>
+                )}
+              </Col>
+              <Col lg={6}>
+                {renderFormGroup("Link to submission", "text", "submissionLink")}
+                {renderFormGroup("Group", "select", "formGroup", "", [
+                  "Please select from the list",
+                  ...activeGroups.map((group) => group.groupName || ""),
+                ])}
+                <Row className="entry-form-button-label d-flex flex-column flex-md-row">
+                  <Col xs="auto">
+                    <Input className="btn-sm entry-form-button" type="submit" value="Submit" />
+                  </Col>
+                  <Col className="pl-0 mt-2 ml-3 mt-md-0">
+                    <Label>
+                      By entering the National Computer Science Competition you agree to the{" "}
+                      <a href="#terms-and-conditions" onClick={handleTermsClick}>
```

Comment:
does this actually link to the right place?

Reply:
Yes, that scrolls the page down to the accordion, and then opens the accordion to Terms and conditions tab